fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt

```diff
@@ -0,0 +1,270 @@
+            draggingItemKey = it.key
+            if (shouldLongPress) {
+                hapticFeedback.performHapticFeedback(HapticFeedbackType.LongPress)
+                onLongPress(it)
+            }
+            draggingItemInitialOffset = it.offset.toFloat()
+            moved = !shouldLongPress
+        }
+    }
+
+    internal fun onDragInterrupted() {
+        if (draggingItemKey != null) {
+            previousKeyOfDraggedItem = draggingItemKey
+            val startOffset = draggingItemOffset
+            scope.launch {
+                previousItemOffset.snapTo(startOffset)
+                previousItemOffset.animateTo(
+                    0f,
+                    spring(
+                        stiffness = Spring.StiffnessMediumLow,
+                        visibilityThreshold = 1f,
+                    ),
+                )
+                previousKeyOfDraggedItem = null
+            }
+        }
+        draggingItemCumulatedOffset = 0f
+        draggingItemKey = null
+        draggingItemInitialOffset = 0f
+    }
+
+    internal fun onDrag(offset: Float) {
+        draggingItemCumulatedOffset += offset
+
+        if (draggingItemLayoutInfo == null) {
+            moved = false
+        }
+        val draggingItem = draggingItemLayoutInfo ?: return
+
+        if (!moved && draggingItemCumulatedOffset > touchSlop) {
+            onExitLongPress()
+        }
+        val startOffset = draggingItem.offset + draggingItemOffset
+        val endOffset = startOffset + draggingItem.size
+        val middleOffset = startOffset + (endOffset - startOffset) / 2f
+
+        val targetItem = listState.layoutInfo.visibleItemsInfo.find { item ->
+            middleOffset.toInt() in item.offset..item.endOffset && draggingItemKey != item.key
+        }
+
+        if (targetItem != null && targetItem.key !in ignoredItems) {
+            if (draggingItem.index == listState.firstVisibleItemIndex) {
+                scope.launch {
+                    listState.scrollBy(-draggingItem.size.toFloat())
+                }
+            }
+            onMove.invoke(draggingItem, targetItem)
+        } else {
+            val overscroll = when {
+                draggingItemCumulatedOffset > 0 ->
+                    (endOffset - listState.layoutInfo.viewportEndOffset).coerceAtLeast(0f)
+
+                draggingItemCumulatedOffset < 0 ->
+                    (startOffset - listState.layoutInfo.viewportStartOffset).coerceAtMost(0f)
+
+                else -> 0f
+            }
+            if (overscroll != 0f) {
+                scope.launch {
+                    listState.scrollBy(overscroll)
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Container for draggable list item
+ *
+ * @param state List reordering state
+ * @param key Key of the item to be displayed
+ * @param content Content of the item to be displayed
+ */
+@ExperimentalFoundationApi
+@Composable
+fun LazyItemScope.DragItemContainer(
```

Comment:
I couldn't find the import for `LazyListItemScope`, I used `LazyItemScope` because it is the one [used](https://developer.android.com/reference/kotlin/androidx/compose/foundation/lazy/LazyItemScope) for lazy row/columns.